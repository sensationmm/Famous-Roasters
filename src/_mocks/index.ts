import { CatalogueMockError, CatalogueMockMissingData, CatalogueMocks, CatalogueMocksPagination } from './CatalogueMock'
import { ProductMock } from './ProductMock'

export { CatalogueMocks as CatalogueMocks }
export { CatalogueMocksPagination as CatalogueMocksPagination }
export { CatalogueMockError as CatalogueMockError }
export { CatalogueMockMissingData as CatalogueMockMissingData }
export { ProductMock as ProductMock }
