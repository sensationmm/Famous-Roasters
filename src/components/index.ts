import { Button, ButtonColor, ButtonEmphasis, ButtonSize } from './Button'
import { Footer } from './Footer'
import { Layout } from './Layout'
import { Navigation } from './Navigation'
import { Typography, TypographySize, TypographyType } from './Typography'

export { Button as Button }
export { ButtonColor as ButtonColor }
export { ButtonEmphasis as ButtonEmphasis }
export { ButtonSize as ButtonSize }
export { Footer as Footer }
export { Layout as Layout }
export { Navigation as Navigation }
export { Typography as Typography }
export { TypographyType as TypographyType }
export { TypographySize as TypographySize }
