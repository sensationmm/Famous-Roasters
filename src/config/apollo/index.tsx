import { ApolloClient, ApolloLink, createHttpLink, from, InMemoryCache, NormalizedCacheObject } from '@apollo/client'
import { onError } from '@apollo/client/link/error'

export const apolloClient = (): ApolloClient<NormalizedCacheObject> => {
  const errorLink = onError(({ graphQLErrors, networkError }) => {
    if (graphQLErrors)
      graphQLErrors.forEach(({ message: errorMessage, locations, path }) => {
        console.log(`[GraphQL error]: Message: ${errorMessage}, Location: ${locations}, Path: ${path}`)
      })

    if (networkError) console.log(`[Network error]: ${networkError}`)
  })

  return new ApolloClient({
    cache: new InMemoryCache(),
    link: from([
      errorLink,
      createHttpLink({
        uri: process.env.REACT_APP_SHOPIFY_STOREFRONT_GRAPHQL_ENDPOINT,
        headers: {
          'X-Shopify-Storefront-Access-Token': process.env.REACT_APP_SHOPIFY_STOREFRONT_ACCESS_TOKEN,
          Accept: 'application/graphql',
        },
        fetch,
      }) as unknown as ApolloLink,
    ]),
  })
}
