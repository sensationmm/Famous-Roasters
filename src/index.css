@tailwind base;
@tailwind components;
@tailwind utilities;

body {
  margin: 0;
  padding: 0;
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

body > div > div > main {
  min-height: 50vh;
}

:root {
  --swiper-theme-color: #646575 !important;
}

.video-container {
  position: relative;
  padding-bottom: 56.25%;
  padding-top: 0;
  height: 0;
  overflow: hidden;
}

.video-container iframe,
.video-container object,
.video-container embed {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}