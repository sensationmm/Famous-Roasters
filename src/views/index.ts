import { About } from './About'
import { Account } from './Account'
import { Cart } from './Cart'
import { Catalogue } from './Catalogue'
import { Checkout } from './Checkout'
import { Contact } from './Contact'
import { Error } from './Error'
import { Home } from './Home'
import { OurRoasters } from './OurRoasters'
import { Product } from './Product'
import { TasteFinder } from './TasteFinder'

export { About as About }
export { Account as Account }
export { Cart as Cart }
export { Catalogue as Catalogue }
export { Checkout as Checkout }
export { Contact as Contact }
export { Error as Error }
export { Home as Home }
export { OurRoasters as OurRoasters }
export { Product as Product }
export { TasteFinder as TasteFinder }
