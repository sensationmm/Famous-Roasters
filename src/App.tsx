import React from 'react'
import { I18nextProvider } from 'react-i18next'
import { BrowserRouter, Route, Routes } from 'react-router-dom'
import {
  About,
  Account,
  Cart,
  Catalogue,
  Checkout,
  Contact,
  Error,
  Home,
  OurRoasters,
  Product,
  TasteFinder,
} from 'src/views'

import { i18n } from './config'

function App() {
  return (
    <I18nextProvider i18n={i18n}>
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/about" element={<About />} />
          <Route path="/account" element={<Account />} />
          <Route path="/cart" element={<Cart />} />
          <Route path="/catalogue" element={<Catalogue />} />
          <Route path="/checkout" element={<Checkout />} />
          <Route path="/contact" element={<Contact />} />
          <Route path="/our-roasters" element={<OurRoasters />} />
          <Route path="/product/:id" element={<Product />} />
          <Route path="/taste-finder" element={<TasteFinder />} />
          <Route path="*" element={<Error />} />
        </Routes>
      </BrowserRouter>
    </I18nextProvider>
  )
}

export default App
