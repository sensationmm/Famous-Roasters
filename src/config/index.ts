import { apolloClient } from './apollo'
import i18n from './i18n'

export { i18n as i18n }
export { apolloClient as apolloClient }
