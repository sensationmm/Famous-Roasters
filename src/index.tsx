import './index.css'

import React from 'react'
import { createRoot } from 'react-dom/client'

import App from './App'

const container = document.getElementById('root') as Element
const root = createRoot(container)
if (process.env.NODE_ENV === 'development') {
  root.render(
    <React.StrictMode>
      <App />
    </React.StrictMode>,
  )
} else {
  root.render(<App />)
}
