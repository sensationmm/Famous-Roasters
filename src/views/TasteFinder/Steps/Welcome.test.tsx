import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import React from 'react'
import { I18nextProvider } from 'react-i18next'
import { i18n } from 'src/config'

import { Welcome } from '.'

global.alert = jest.fn()

describe('Welcome step partial view', () => {
  it('Renders correctly', async () => {
    const { container } = render(
      <I18nextProvider i18n={i18n}>
        <Welcome next={() => null} />
      </I18nextProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('The user can click on start', async () => {
    render(
      <I18nextProvider i18n={i18n}>
        <Welcome next={() => alert('start!')} />
      </I18nextProvider>,
    )
    const startButton = await screen.findByTestId('button-start')
    expect(startButton).toBeInTheDocument()
    fireEvent.click(startButton)
  })
})
