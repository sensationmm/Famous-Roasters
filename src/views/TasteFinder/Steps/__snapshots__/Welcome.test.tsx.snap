// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome step partial view Renders correctly 1`] = `
<div>
  <div
    class="flex flex-col md:grid md:grid-cols-3 md:grid-rows-1"
    style="height: calc(100vh - 66px);"
  >
    <div
      class="flex items-center justify-center py-24 mt-4 relative md:justify-start md:order-2 md:py-8 md:mt-0"
    >
      <img
        alt="Tastefinder"
        class="absolute top-16 max-h-64 md:top-auto md:max-h-96 md:-left-16 xl:max-h-96 xl:-left-32"
        src="welcome.webp"
      />
    </div>
    <div
      class="grid items-center justify-center pt-10 pb-44 bg-brand-black text-white md:order-1 md:py-8 md:grid-cols-10 md:col-span-2 xl:grid-cols-5"
    >
      <div
        class="flex flex-col col-items-center md:col-start-2 md:col-end-7 md:order-1 xl:col-start-2 xl:col-end-4"
      >
        <h1
          class="flex w-80 mt-32 font-syne md:mt-0 text-3xl leading-10 font-semibold tracking-tight"
        >
          Finde jetzt deinen Specialty Coffee mit dem Tastefinder.
        </h1>
        <div
          class="flex w-80 mt-4"
        >
          <svg
            aria-hidden="true"
            class="w-12 h-6 mr-4"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M12.293 5.293a1 1 0 011.414 0l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-2.293-2.293a1 1 0 010-1.414z"
              fill-rule="evenodd"
            />
          </svg>
          <p
            class="text-base leading-6"
          >
            Finde deinen pers√∂nlichen Kaffeetyp und erhalte Empfehlungen, die zu deinem Geschmack passen.
          </p>
        </div>
        <div
          class="flex w-80 mt-8 xl:mt-24"
        >
          <button
            class="flex w-full justify-center inline-flex items-center rounded-full border-2 px-4 py-2 text-base font-bold disabled:bg-brand-grey-whisper disabled:text-coreUI-text-secondary disabled:cursor-not-allowed text-brand-black bg-primary border-transparent hover:bg-brand-green-starship active:bg-primary"
            data-testid="button-start"
            role="button"
            type="button"
          >
            Tastefinder starten
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
